modules/service/src/main/scala/lucuma/odb/graphql/mapping/ResultMapping.scala

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2016-2022 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.odb.graphql.mapping
+
+import cats.syntax.all._
+import edu.gemini.grackle.Cursor
+import edu.gemini.grackle.Cursor.Env
+import edu.gemini.grackle.Cursor.ListTransformCursor
+import edu.gemini.grackle.Mapping
+import edu.gemini.grackle.Query
+import edu.gemini.grackle.Query.*
+import edu.gemini.grackle.Result
+import edu.gemini.grackle.Type
+import io.circe.Encoder
+import lucuma.odb.graphql.BaseMapping
+import skunk.codec.numeric.int8
+
+object ResultMapping {
+
+  // keys for things we expect to find in the environment
+  val LimitKey = "resultLimit"
+  val AliasKey = "resultAlias"
+
+  private def hasMore(collectionField: String)(c: Cursor): Result[Boolean] =
+    for
+      limit <- c.envR[Int](LimitKey)
+      alias <- c.envR[Option[String]](AliasKey)
+      items <- c.field(collectionField, alias)
+      size  <- items.listSize
+    yield limit < 0 || size > limit
+
+
+  /** A cursor transformation that takes `n` elements from a list (if it's a list). */
+  object Take {
+    def apply(n: Int)(c: Cursor): Result[Cursor] =
+      (c.listSize, c.asList(Seq)).mapN { (size, elems) =>
+        if size <= n then c
+        else ListTransformCursor(c, size - 1, elems.init)
```

Comment:
This is where the extra item for `hasMore` is taken out?

Reply:
Correct.